### ISSUE TITLE
Porting to SDL2

### ISSUE THREAD

**Issue by ghost** [VenKamikaze/BitRiot#7]
Hi,

I've done basic porting to SDL2. 

The game still renders all things to a SDL_Surface (sdl_primary).
I just modiefied the code to work with the needed SDL_Texture and SDL_Renderer
and SDL2 in general (changed functions etc..)

It doesn't use Textures instead of Surfaces to load the bitmaps and so on.
That would need some more work / rewriting things.

As far as I can see the changes shouldn't break anything or change game behavior.

If you are interested I could upload the changes to a new branch of my fork. 
(If you plan to merge PR https://github.com/VenKamikaze/BitRiot/pull/5 I would wait until you merged it)

Let me know...

**Comment by VenKamikaze** [comment]
Hey, that sounds great. I'll be interested to see the changes!

I've merged PR #5 now too.

**Comment by ghost** [comment]
I've uploaded it to the sdl2 branch of my fork.
I hope I didn't miss anything. ;)

**Comment by VenKamikaze** [comment]
I haven't had a chance to look at this yet but hopefully in the next week or so I will get some time. I'm really pleasantly surprised by the frequency of pull requests I've been getting lately! 

I was thinking of making an sdl2 branch here as well to pull in your changes and perhaps continue working on it to use textures for the bitmaps. I would merge it to master after a short while (whether or not there had been additional sdl2 work) and continue any sdl2 related development in its own branch with occasional pulls back into master. What do you think?

**Comment by ghost** [comment]
That sounds good to me.

**Comment by VenKamikaze** [comment]
I've pulled in your changes to an sdl2 branch on my own repo. Thanks for that!

I've branched from there with 'sdl2_ongoing' where I'm making some further changes. 

I'm also just getting my head around the new SDL_Renderer & SDL_Texture stuff and I've been reading this migration guide: https://wiki.libsdl.org/MigrationGuide . Looks like there still needs to be a few changes to switch everything over to using Textures and RenderCopy, and moving away from Blitting. It might take me a while to get through it, but it might be an interesting enough learning experience :)

Also, were you still making changes on your sdl2 branch? I just want to make sure I don't overlap if you are.

**Comment by ghost** [comment]
Hi, thanks for using my sdl2 branch. :)

> Also, were you still making changes on your sdl2 branch? I just want to make sure I don't overlap if you are. 

No, the last change I made was removing the DSound code.

**Comment by VenKamikaze** [comment]
I've merged your PR #14 .

I've also decided I shouldn't hold up getting all these improvements that have been made in the sdl2 branches into master any longer, so I'm pulling sdl2_ongoing into sdl2 and then into master, will tag the current code as an sdl2 release once there. 

I may continue with the switch to SDL_Texture instead of most surfaces at a later date, depending on time.

I'll close this issue when I've merged everything up and tagged the sdl2 release.

Thanks for all of your contributions!
